src/auto/converter.rs

```diff
@@ -0,0 +1,48 @@
+// This file was generated by gir (https://github.com/gtk-rs/gir)
+// from gir-files (https://github.com/gtk-rs/gir-files)
+// DO NOT EDIT
+
+use ConverterFlags;
+use ConverterResult;
+use Error;
+use ffi;
+use glib::object::IsA;
+use glib::translate::*;
+use glib_ffi;
+use gobject_ffi;
+use std::mem;
+use std::ptr;
+
+glib_wrapper! {
+    pub struct Converter(Object<ffi::GConverter, ffi::GConverterIface>);
+
+    match fn {
+        get_type => || ffi::g_converter_get_type(),
+    }
+}
+
+pub trait ConverterExt {
+    fn convert(&self, inbuf: &[u8], outbuf: &[u8], flags: ConverterFlags) -> Result<(ConverterResult, usize, usize), Error>;
+
+    fn reset(&self);
+}
+
+impl<O: IsA<Converter>> ConverterExt for O {
+    fn convert(&self, inbuf: &[u8], outbuf: &[u8], flags: ConverterFlags) -> Result<(ConverterResult, usize, usize), Error> {
+        let inbuf_size = inbuf.len() as usize;
+        let outbuf_size = outbuf.len() as usize;
+        unsafe {
```

Comment:
As I remember `outbuf` need be mutable as converter write data to it,
something like `buffer` in `InputStreamExtManual::read`,
`inbuf` also better be `AsRef<[u8]>` as in `FileExtManual::replace_contents_async` but maybe it unneeded.

Reply:
Yes, at least make it a `&mut [u8]` but otherwise `AsRef<[u8]>` / `AsMut<[u8]>` would be more generic. I'd suggest using references though as this function does not take ownership of the objects, and the user can always get the slice reference themselves